spacy/cli/apply.py

```diff
@@ -0,0 +1,141 @@
+import tqdm
+import srsly
+
+from itertools import chain
+from pathlib import Path
+from typing import Optional, List, Iterable, cast, Union
+
+from wasabi import msg
+
+from ._util import app, Arg, Opt, setup_gpu, import_code, walk_directory
+
+from ..tokens import Doc, DocBin
+from ..vocab import Vocab
+from ..util import ensure_path, load_model
+
+
+path_help = (
+    "Location of the documents to predict on. "
+    "Can be a single file in .spacy format or a "
+    ".jsonl file. Files with other extensions "
+    "are treated as single plain text documents. "
+    "If a directory is provided "
+    "it is traversed recursively to grab all files to "
+    "be processed. The files can be a mixture of .spacy, "
+    ".jsonl and text files. If .jsonl is provided the "
+    "specified field is going to be grabbed ('text' "
+    "by default)."
+)
+out_help = "Path where to save the result .spacy file"
+code_help = (
+    "Path to Python file with additional " "code (registered functions) to be imported"
+)
+gold_help = "Use gold preprocessing provided in the .spacy files"
+force_msg = (
+    "The provided output file already exists. "
+    "To force overwriting the output file, set the --force or -F flag."
+)
+
+
+DocOrStrStream = Union[Iterable[str], Iterable[Doc]]
+
+
+def _stream_docbin(path: Path, vocab: Vocab) -> Iterable[Doc]:
+    """
+    Stream Doc objects from DocBin.
+    """
+    docbin = DocBin().from_disk(path)
+    for doc in docbin.get_docs(vocab):
+        yield doc
+
+
+def _stream_jsonl(path: Path, field: str) -> Iterable[str]:
+    """
+    Stream "text" field from JSONL. If the field "text" is
+    not found it raises error.
+    """
+    for entry in srsly.read_jsonl(path):
+        if field not in entry:
+            raise msg.fail(
+                f"{path} does not contain the required '{field}' field.", exits=1
+            )
```

Comment:
`msg.fail` doesn't create an exception, so it doesn't make sense to raise it. Also with `exits=1` it'll exit the program, so you can't raise anyway.